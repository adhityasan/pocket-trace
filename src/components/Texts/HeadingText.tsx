import { useMemo } from "react";
import { StyleSheet, Text, TextProps } from "react-native";

import { COLOR_LIGHT } from "@services/constants/color";

interface IHeadingTextProps extends TextProps {
  type: "h1" | "h2" | "h3" | "h4";
}

const HeadingText = ({ type, style, children }: IHeadingTextProps) => {
  const textStyle = useMemo(
    () =>
      StyleSheet.flatten([
        styles.default,
        type === "h1" && styles.h1,
        type === "h2" && styles.h2,
        type === "h3" && styles.h3,
        type === "h4" && styles.h4,
        style
      ]),
    [type, style]
  );

  return <Text style={textStyle}>{children}</Text>;
};

const styles = StyleSheet.create({
  default: {
    fontWeight: "bold",
    marginBottom: 3,
    color: COLOR_LIGHT.PRIMARY
  },
  h1: {
    fontSize: 25
  },
  h2: {
    fontSize: 20
  },
  h3: {
    fontSize: 18
  },
  h4: {
    fontSize: 15
  }
});

export default HeadingText;
