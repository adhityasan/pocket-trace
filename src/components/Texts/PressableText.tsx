import {
  Text,
  TouchableOpacityProps,
  TouchableOpacity,
  StyleSheet,
  StyleProp,
  TextStyle
} from "react-native";

import { COLOR_LIGHT } from "@services/constants/color";

interface IPressableTextProps extends TouchableOpacityProps {
  children: JSX.Element | string;
  style?: StyleProp<TextStyle>;
  wrapperStyle?: StyleProp<TextStyle>;
}

const PressableText = ({
  children,
  style,
  wrapperStyle,
  ...touchableOpacityProps
}: IPressableTextProps) => {
  return (
    <TouchableOpacity {...touchableOpacityProps} style={wrapperStyle}>
      <Text style={[styles.text, style]}>{children}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  text: {
    color: COLOR_LIGHT.PRIMARY
  }
});

export default PressableText;
