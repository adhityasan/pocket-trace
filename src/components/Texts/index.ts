export { default as HeadingText } from "./HeadingText";
export { default as PressableText } from "./PressableText";
