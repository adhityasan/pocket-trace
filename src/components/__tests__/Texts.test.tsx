import { create } from "react-test-renderer";

import { HeadingText, PressableText } from "@components/Texts";

/**
 * @components/Inputs/TextInput.tsx
 */
describe("components/Texts/HeadingText", () => {
  it("renders correctly", () => {
    const headingTextRenderer = create(<HeadingText type="h1">Heading Text</HeadingText>).toJSON();
    expect(headingTextRenderer).toMatchSnapshot();
  });
});

describe("components/Texts/PressableText", () => {
  it("renders correctly", () => {
    const pressableTextRenderer = create(<PressableText>Pressable Text</PressableText>).toJSON();
    expect(pressableTextRenderer).toMatchSnapshot();
  });

  it("trigger action on press", () => {
    let isPressed = false;
    const pressableTextRenderer = create(
      <PressableText
        onPress={() => {
          isPressed = true;
        }}>
        Pressable Text
      </PressableText>
    );
    const pressableTextInstance = pressableTextRenderer.root;
    pressableTextInstance.props.onPress();
    expect(isPressed).toBeTruthy();
  });
});
