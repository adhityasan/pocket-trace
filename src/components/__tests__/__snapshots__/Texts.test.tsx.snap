// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Texts/HeadingText renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#5145a0",
      "fontSize": 25,
      "fontWeight": "bold",
      "marginBottom": 3,
    }
  }
>
  Heading Text
</Text>
`;

exports[`components/Texts/PressableText renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#5145a0",
        },
        undefined,
      ]
    }
  >
    Pressable Text
  </Text>
</View>
`;
